libvast/src/system/importer.cpp

```diff
@@ -40,23 +40,33 @@ namespace vast::system {
 
 namespace {
 
-class driver
-  : public caf::stream_stage_driver<
-      table_slice,
-      caf::broadcast_downstream_manager<detail::framed<table_slice>>> {
+class driver : public caf::stream_stage_driver<
+                 table_slice, caf::broadcast_downstream_manager<table_slice>> {
 public:
-  driver(caf::broadcast_downstream_manager<detail::framed<table_slice>>& out,
+  driver(caf::broadcast_downstream_manager<table_slice>& out,
          importer_state& state)
     : stream_stage_driver(out), state{state} {
     // nop
   }
 
-  void process(caf::downstream<detail::framed<table_slice>>& out,
+  void process(caf::downstream<table_slice>& out,
                std::vector<table_slice>& slices) override {
     VAST_TRACE_SCOPE("{}", VAST_ARG(slices));
     uint64_t events = 0;
     auto t = timer::start(state.measurement_);
     for (auto&& slice : std::exchange(slices, {})) {
+      auto import_ok = state.executor.add(std::move(slice));
+      if (import_ok) {
+        VAST_WARN("importer can not add slice to pipeline executor - data "
+                  "loss");
+      }
+    }
+    auto transformed_slices = state.executor.finish();
+    if (!transformed_slices) {
```

Comment:
Should we continue in such case? Later on the transformed_slices are dereferenced. I think this would crash when bool operator fails for caf::expected.

Reply:
I did mean the keyword continue :D
The else achieves the same, maybe it is a bit less readable for me but i am fine with both.